File: find_start_end_signal.m
function y = find_start_end_signal()
    f1 = fopen('sixtythousandbits.dat', 'r');
    tmp = fread(f1,'float32');
    fclose(f1);
    y = tmp(1:2:end)+1i*tmp(2:2:end);
    
    real_y = real(y); % getting the real part of the y vector
    plot(real_y);
    
    length(real_y)
    
    start_of_signal = 0; % going to be the first index
    for i=2000:length(real_y)
      if (abs(real_y(i)) > 0.002)
        start_of_signal = i-5
        break
      end
    end
    
    end_of_signal = length(real_y);
    for i = length(real_y):-1:5
      if (abs(real_y(i)) > 0.002)
        end_of_signal = i+5
        break
      end
    end
    plot(real_y(start_of_signal:end_of_signal));
end

File: find_start_end_signal.py
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
from decompress import *
import data_processing


def find_start_end_signal(filename):
    y = read_from_file(filename)
    y = data_processing.decomplexize_data(y)

    # get real component of y
    print "complex y ", y
    real_y = [float(x) for x in y]
    print "real_y ", real_y[0], real_y[1]

    # get start & end of signal
    start_of_signal = 0 # going to be first index at first (update with later one if I have time)
    for i in range(len(real_y)):
        if (abs(real_y[i]) > 0.0025):
            start_of_signal = i - 3
            break


    end_of_signal = len(real_y)-1
    for i in list(reversed(range(len(real_y)))):
        if (abs(real_y[i]) > 0.0025):
            end_of_signal = i + 3
            break

    print "start & end of signal: ", (start_of_signal, end_of_signal)
    # plt.plot(real_y[start_of_signal:end_of_signal])
    # plt.show()
    return (start_of_signal, end_of_signal)

# find_start_end_signal()